### PR TITLE
[WIP] Alternate Evolve impl

### PR DESCRIPTION
**Pull request by devinsba**
Attempting to get #347 working more cleanly with the API and a few things I wanted

### CODE REVIEW

File: src/main/kotlin/ink/abb/pogo/scraper/tasks/SmartEvolve.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * Created by ddcbdevins on 7/26/16.
+ */
+class SmartEvolve : Task {
+    override fun run(bot: Bot, ctx: Context, settings: Settings) {
+        val pokebagFillPercent = ctx.api.inventories.pokebank.pokemons.size.toDouble() / ctx.profile.pokemonStorage
+        Log.white("Pokebag ${pokebagFillPercent * 100} % full.")
+        val groupedPokemon = ctx.api.inventories.pokebank.pokemons.groupBy { it.pokemonId }
+        if (pokebagFillPercent >= 0.7) {
```
Comment (devinsba) [inline src/main/kotlin/ink/abb/pogo/scraper/tasks/SmartEvolve.kt]: This value should be a setting

**Comment by jabbink** [comment]
FYI regarding Eevee evolves

http://gamerant.com/pokemon-go-guide-eevee-evolution-flareon-vaporeon-jolteon/

tl;dr

Rename pokemon:

* Sparky = Jolteon
* Rainer = Vaporeon
* Pyro = Flareon

**Comment by marcx093** [comment]
Eevee easter egg only works first time for every pokemon. For example you can't get more than 1 vaporeon with 100% accuracy.

**Comment by jabbink** [comment]
Good to know.

**Comment by megatixos** [comment]
Why dont you merge this already @jabbink ? What is the problem holding it back?

**Comment by joligario** [comment]
> Why dont you merge this already @jabbink ? What is the problem holding it back?

I would guess because it is a Work In Progress (WIP). Why don't you pull it down on your own, check it out, and report back if it is operational?

**Comment by devinsba** [comment]
I had been running it until the API lockdown. I have some code I haven't
pushed up and it isn't configurable at all for and end user.

File: src/main/kotlin/ink/abb/pogo/scraper/Bot.kt [hunk 2]
```diff
@@ -112,8 +113,11 @@ class Bot(val api: PokemonGo, val settings: Settings) {
                 task(catch)
             if (settings.shouldDropItems)
                 task(drop)
-            if (settings.shouldAutoTransfer)
+            if (settings.shouldAutoTransfer) {
                 task(release)
+            } else {
+                task(smartEvolve)
```
Comment (jabbink) [inline src/main/kotlin/ink/abb/pogo/scraper/Bot.kt]: Why don't allow evolving and transferring as well?

Comment (devinsba) [inline src/main/kotlin/ink/abb/pogo/scraper/Bot.kt]: It happened to fit what I was planning on at the time. I am considering the final touches to make this all configurable now.

**Comment by devinsba** [comment]
@jabbink should I do a rebase on this to clean up the commit history?

File: src/main/kotlin/ink/abb/pogo/scraper/tasks/Evolve.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import ink.abb.pogo.scraper.evolve.EvolutionStrategy
+import ink.abb.pogo.scraper.evolve.IvMaximizingStrategy
+import ink.abb.pogo.scraper.util.Log
+
+/**
+ * Created by ddcbdevins on 7/26/16.
+ */
+class Evolve : Task {
+
+    private val DEFAULT_EVOLUTION_STRATEGY = "max_iv"
```
Comment (devinsba) [inline src/main/kotlin/ink/abb/pogo/scraper/tasks/Evolve.kt]: The implementation in #939 should be the default implementation but I just put mine in here as a placeholder

**Comment by devinsba** [comment]
@jabbink this is no longer WIP status

**Comment by devinsba** [comment]
@jabbink since #939 was merged in, should I refactor this around that implementation?

**Comment by staurakis** [comment]
i think there is a logic error in evolving.....
Firstly there should be an option/setting to be able to evolve automatically and be able to keep the evolved pokemon based on iv stats. In other words to be able to say evolve only pokemons above certain IV and KEEP them!
Secondly, check the message bellow i received:
13 Αυγ 11:49:29 [default: ProfileLoop] - Going to transfer ARBOK with CP 797 and IV 93%; reason: Evolved for XP only
After the evolve, a pokemon that shouldn't be transferred was transferred. A pokemon with IV above 90%.
It wasn't in obligatory transfer or anything like that. 
By the way, if i may comment this version i am testing....it is THE GREATEST without any of the known issues.....no authentication issues, no getting the map and not doing anything...perfect in everything! :)
